### DISCUSSION TITLE
Install toltec and toltec packages offline via scp?

### DISCUSSION THREAD

**Discussion by glyh** [toltec-dev/toltec#625] [Support] [answered]
Hello all, my remarkable 2 tablet is having trouble to install toltec by directly running the command given in the README, so I wonder if it's possible to install via scp over ssh? Thank you!

**Comment by Eeems** [comment, marked as answer]
That would require a lot more work than I would consider worth it. Since toltec expects an internet connection to even be useful, I think it's a better idea to sort out why the installation is failing for you. What issues are you encountering?

**Reply by glyh** [reply]
I can ping google but I can't ping toltec-dev.org. I can't open the website. Is there any mirror sites? Or how can I create one?

BTW why not just host them on github or some other places?

**Reply by LinusCDE** [reply]
The files are actually just build with GitHubs actions and then uploaded to the server hosting it.

To be recognized as a valid repo, your Webserver needs a directory that contains the files `Packages` and `Packages.gz`. These files contain basic package info and also the filename of the packages present in the same directory.

So in the case of toltec, we have 3 of those repos (for each arch):
 - https://toltec-dev.org/stable/rmall/
 - https://toltec-dev.org/stable/rm1/
 - https://toltec-dev.org/stable/rm2/

The repos are defined in `/opt/etc/opkg.conf` and can just be pointed to your webserver containing a mirror.

As for why no offline, I believe, that we incooperate the default installer for opkg (some script called `generic.sh`) which downloads a `opkg` binary and some specific files and packages to get the package manager started. Translating that to offline would be a lot of hassle and have us test both scenarios.

My advice is to install it using the bootstrap script. After that however, you can probably just install `.ipk` files manually if needed. The files are generated on each pull request. They only include changed files (so a PR to update a package only builds that package). All the stable packages are merged in weekly pull requests. So those will contain all the update. Once one is merged, you can grab all the updates like so:

![screenshot5351](https://user-images.githubusercontent.com/22298664/188516052-989cd1e2-fa5a-4345-8313-bb2339da7f58.png)

Keep in mind that we may refer to [entware packages](https://bin.entware.net/armv7sf-k3.2/Packages.html), or expect updates of those as well. So be sure to also update those.

**Reply by glyh** [reply]
@LinusCDE You can't do that without connection to toltec-dev.org, bootstrap script is trying to connect toltec-dev.org. It's trying to bootstrap some softwares, e.g. wget.

EDIT: I just manage to use my phone as hotspot to download files from toltec-dev.org, so probably this is not a big issue at the moment.

**Reply by LinusCDE** [reply]
Ah sorry. Assumed you needed offline functionality past the install with bootstrap.

**Reply by Eeems** [reply]
> @LinusCDE You can't do that without connection to toltec-dev.org, bootstrap script is trying to connect toltec-dev.org. It's trying to bootstrap some softwares, e.g. wget.
> 
> EDIT: I just manage to use my phone as hotspot to download files from toltec-dev.org, so probably this is not a big issue at the moment.

Are you able to connect to toltec-dev.org from a computer on the same network? What happens if you `nslookup toltec-dev.org` from your reMarkable?
